src/AudioData.h

```diff
@@ -0,0 +1,44 @@
+#ifndef AUDIO_DATA_HPP
+#define AUDIO_DATA_HPP
+
+#include <stdint.h>
+extern "C" {
+  #include "goom.h"
+  #include "goom_config.h"
+}
+
+
+static inline int16_t FloatToInt16(float f)
+{
+  if (f >= 1.0f)
+    return INT16_MAX;
+  else if (f < -1.0f)
+    return -INT16_MAX;
+  else
+    return static_cast<int16_t>((f * static_cast<float>(INT16_MAX)));
+}
+
+static inline void FillAudioDataBuffer(int16_t audioData[NUM_AUDIO_SAMPLES][AUDIO_SAMPLE_LEN], 
+  const float floatAudioData[NUM_AUDIO_SAMPLES*AUDIO_SAMPLE_LEN], int numChannels)
+{
+  if (numChannels == 1) 
+  {
+    for (int i=0; i < AUDIO_SAMPLE_LEN; i++) 
+    {
+      audioData[0][i] = FloatToInt16(floatAudioData[i]);
+      audioData[1][i] = audioData[0][i];
+    }
+  } else 
+  {
+    int fpos = 0;
+    for (int i=0; i < AUDIO_SAMPLE_LEN; i++) 
```

Comment:
you could use i < AUDIO_SAMPLE_LEN / 2 and increase i by 2

Reply:
Sorry, not sure what you mean. I was trying to keep this as simple as possible for clarity. Every second floatAudioData element goes to audioData[1][i]. I'm happy to change it if you show me the loop you mean.

Reply:
Yes, sure:

`for (int i = 0; i < AUDIO_SAMPLE_LEN / 2; i += 2)
{
  audiodata[0][i] = whatever[i];
  audiodate[1][i] = whatever[i+1];
}`

Your code is fine - just decide what is easier.

Reply:
You decide. If you say, your's is easier, please keep it and btw. thanks that you look into that stuff.

@AlwinEsch  please have a look at the locking.

Reply:
Also a good why by yours @fritsch